src/features/profile/components/Modals/AchievementDetailsModal.tsx

```diff
@@ -16,10 +16,14 @@ export const AchievementDetailsModal = ({ visible, hideModal, id }: Props) => {
   const achievement = useAchievementDetails(id)
   const theme = useTheme()
 
+  useEffect(() => {
+    if (!visible) return
+    achievement?.track()
+  }, [achievement, visible])
```

Comment:
Est-ce que c'est mieux comme ça ? 

```suggestion
  useEffect(() => {
    if (visible) {
    return achievement?.track()
    }
  }, [achievement, visible])
```

Reply:
Je suis un never nester compulsif mais oui ta suggestion fonctionne aussi. https://www.martin-paucot.fr/blog/stop-nesting-your-code-4ng8